src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts

```diff
@@ -53,25 +53,29 @@ export class TextCellComponent extends CellView implements OnInit, OnChanges {
 		this._activeCellId = value;
 	}
 
-	@HostListener('document:keydown.escape', ['$event'])
-	handleKeyboardEvent() {
-		if (this.isEditMode) {
-			this.toggleEditMode(false);
-		}
-		this.cellModel.active = false;
-		this._model.updateActiveCell(undefined);
-	}
-
 	// Double click to edit text cell in notebook
 	@HostListener('dblclick', ['$event']) onDblClick() {
 		this.enableActiveCellEditOnDoubleClick();
 	}
 
 	@HostListener('document:keydown', ['$event'])
 	onkeydown(e: KeyboardEvent) {
+		const keyEvent = new StandardKeyboardEvent(e);
+		if (this.cellModel?.currentMode !== CellEditModes.CODE && this.cellModel.active) {
+			if (keyEvent.keyCode === KeyCode.Escape) {
+				if (this.isEditMode) {
+					this.toggleEditMode(false);
+					e.stopImmediatePropagation();
+				}
+			} else if (keyEvent.keyCode === KeyCode.Enter) {
+				this.toggleEditMode(true);
+				this.cellModel.active = true;
```

Comment:
This seems like it should go into `updateActiveCell` where that will set active to true for whatever cell model is being set to the new active one

Reply:
the updateActiveCell method already sets active to true. Removing this logic